web/cobrands/fixmystreet/assets.js

```diff
@@ -416,28 +393,62 @@ fixmystreet.add_assets = function(options) {
                 }
                 zoom_to_assets(asset_layer);
             } else {
-                asset_layer.setVisibility(false);
+                if (!asset_layer.fixmystreet.always_visible) {
+                    asset_layer.setVisibility(false);
+                }
                 if (asset_fault_layer) {
                     asset_fault_layer.setVisibility(false);
                 }
             }
         });
-    }
 
-    // Make sure the assets get hidden if the back button is pressed
-    fixmystreet.maps.display_around = (function(original) {
-        function hide_assets() {
-            asset_layer.setVisibility(false);
-            if (asset_fault_layer) {
-                asset_fault_layer.setVisibility(false);
+
+        // Make sure the assets get hidden if the back button is pressed
+        fixmystreet.maps.display_around = (function(original) {
+            function hide_assets() {
+                for (var i = 0; i < fixmystreet.assets.layers.length; i++) {
+                    var layer = fixmystreet.assets.layers[i];
+                    if (!asset_layer.fixmystreet.always_visible) {
+                        asset_layer.setVisibility(false);
+                    }
+                }
+                fixmystreet.markers.setVisibility(true);
+                original.apply(fixmystreet.maps);
+            }
+            return hide_assets;
+        })(fixmystreet.maps.display_around);
+    },
+
+    init: function() {
+        if (fixmystreet.map === undefined) {
```

Comment:
Do we want the check-fixmystreet-is-defined safety check to do nothing at all? Or do we think that's okay now.

Reply:
Looks like this is still required, due to the order that things are loaded in. Ideally there'd be a post-setup hook for `fixmystreet.map`, I'll see if that's a simple thing to add.